### PR TITLE
Creating a callback on inventory_type to create tallies

### LINKED ISSUE #454: After creating a new inventory_type, we should get a new inventory tally at all storage locations
There is now a callback on location to create inventory_tallies for all inventory_types after create. 

Following the same pattern, we need an after_create callback on inventory_type to create tallies on all locations that are storage units.

### PR DESCRIPTION
**Pull request by juliafurlani**
Resolves #454

### Description

Creates an after_create callback on inventory_type to create tallies on all locations that are storage units.

### Type of change

- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

Created a spec for inventory_type.

### CODE REVIEW

**Comment by renatamarques97** [comment]
LGTM

File: app/models/inventory_type.rb [hunk 1]
```diff
@@ -2,4 +2,15 @@ class InventoryType < ApplicationRecord
...
   has_many :inventory_tallies
+
+  after_create :create_inventory_tallies
+
+  private
+
+  def create_inventory_tallies
+    locations = Location.where(location_type: "storage_unit")
+    locations.each do |location|
+      inventory_tallies.create(storage_location: location)
```
Comment (maebeale) [inline app/models/inventory_type.rb]: Thanks for working on this, @juliacfurlani !!!

We'll need to do one more loop here so each Location gets an InventoryTally for each InventoryType.

(If this codespeak is helpful: under `locations.each do`, we need `InventoryType.all.each do` and then the `inventory_tallies.create` can include the second param of `inventory_type: inventory_type`)

Would you be willing to make that change and update the tests?

Comment (juliacfurlani) [reply app/models/inventory_type.rb]: Sure, no problem!!

File: spec/models/inventory_type_spec.rb [hunk 2]
```diff
@@ -1,5 +1,35 @@
 require 'rails_helper'
 
 RSpec.describe InventoryType, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  it { 
+    is_expected.to have_many(:box_items)
+    .class_name('BoxItem'),
+    foreign_key: 'inventory_type_id',
+    inverse_of: :assembly_location
+  }
```
Comment (sobrinho) [inline spec/models/inventory_type_spec.rb:9]: Is this syntax correct? It doesn't seem to be.

Try changing the foreign key to another value and see if the spec will fail.

**Review by maebeale — APPROVED** [review]
THANK YOU!!!

**Pull request merged**